addons/popochiu/engine/objects/walkable_area/popochiu_walkable_area.gd

```diff
@@ -32,110 +32,287 @@ extends Node2D
+		return
+
+	# Create a clean navigation polygon without obstacles
+	var clean_navpoly = NavigationPolygon.new()
+	clean_navpoly.agent_radius = 0.0
+	clean_navpoly.cell_size = _perimeter.navigation_polygon.cell_size if _perimeter.navigation_polygon else 1.0
+
+	# Add the original outlines
+	for outline in interaction_polygon:
+		clean_navpoly.add_outline(outline)
+
+	# Bake the clean navigation polygon
+	NavigationServer2D.bake_from_source_geometry_data(clean_navpoly, NavigationMeshSourceGeometryData2D.new())
+
+	# Set up the region with the clean navigation polygon
+	NavigationServer2D.region_set_navigation_polygon(region_rid_no_obstacles, clean_navpoly)
+	NavigationServer2D.region_set_transform(region_rid_no_obstacles, global_transform)
+
+	# Ensure the region is enabled
+	NavigationServer2D.region_set_enabled(region_rid_no_obstacles, enabled)
+
+	# Force update the navigation map without obstacles
+	if NavigationServer2D.map_is_active(map_rid_no_obstacles):
+		NavigationServer2D.map_force_update(map_rid_no_obstacles)
+
+
+#region Public ####################################################################################
+## Collects all the [NavigationObstacle2D]s in the scene and returns them.
+
+## Sets up navigation obstacles using projected obstructions.
+## This is the preferred method for carving out areas in a NavigationPolygon.
+func setup_obstacles(obstacles: Array[NavigationObstacle2D]) -> void:
+	if not _perimeter or not _perimeter is NavigationRegion2D:
+		return
+
+	_load_navigation_polygon()
+
+	# Create source geometry data for baking
+	var source_geometry = NavigationMeshSourceGeometryData2D.new()
+
+	# Now add each obstacle as a projected obstruction
+	for obstacle in obstacles:
+		if not obstacle or obstacle.vertices.size() < 3:
+			continue
+
+		var obstacle_parent = obstacle.get_parent()
+		if not obstacle_parent or not obstacle_parent.visible or not obstacle_parent.obstacle:
+			continue
+
+		# Convert obstacle vertices to global space, then to perimeter's local space
+		var local_vertices = PackedVector2Array()
+		for vertex in obstacle.vertices:
+			# First convert to global space
+			var global_pos = obstacle_parent.to_global(vertex)
+			# Adjust the global position to account for the baseline because
+			# the engine is currently using y-sorting and not z-index.
+			global_pos.y -= obstacle_parent.baseline * scale.y
+			# Then convert to perimeter's local space
+			local_vertices.append(_perimeter.to_local(global_pos))
+
+		# Add as projected obstruction (true means carving)
+		source_geometry.add_projected_obstruction(local_vertices, true)
+
+	_bake_navigation(source_geometry)
+
+	# Wait a frame to ensure navigation is properly updated
+	await get_tree().process_frame
 
 
 #endregion
 
 #region SetGet #####################################################################################
 func _set_enabled(value: bool) -> void:
+	# Always store the value first
```

Comment:
Just to mention that the SetGet region should go between the Public and the Private regions.

Reply:
Came from moving the private region at the bottom.